control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2660,6 +2660,19 @@ func (r *HostedControlPlaneReconciler) reconcileKubeAPIServer(ctx context.Contex
 		return fmt.Errorf("failed to reconcile api server config: %w", err)
 	}
 
+	if !util.HCPOAuthEnabled(hcp) &&
+		len(hcp.Spec.Configuration.Authentication.OIDCProviders) > 0 {
+		authConfig := manifests.AuthConfig(hcp.Namespace)
+		if _, err := createOrUpdate(ctx, r, authConfig, func() error {
+			return kas.ReconcileAuthConfig(authConfig,
+				p.OwnerRef,
+				p.ConfigParams(),
```

Comment:
Are these config params already validated? E.g. you can reach all of the issuers from the OIDCProviders and their discovery returns valid data (per https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata), CA can verify the serving cert at the issuer URL, ...?

Can you point me to the validation code?

Reply:
They weren't validated before either, right?  We probably validate in self-managed, but most of the guide rails are missing from the re-implementation.